### DISCUSSION TITLE
[a] autosave help

### DISCUSSION THREAD

**Discussion by ProfStack** [artisan-roaster-scope/artisan#1090] [Q&A]
The laptop computer running the Hottop 2K+ roast suddenly turned off (MY fault for not watching the battery indicator).  This caused the Hottop to stop mid first crack.  I ejected the beans and all was at best okay.  The data collected was lost when I charged the laptop battery and turned on the laptop.

My question is about the [a] autosave feature.  Does it just save at the end of the roast or can it save during the roast, in case of what happened to me?  TIA.

**Comment by roasterdave** [comment]
Pressing the 'a' key anytime during a roast will save the profile to a file  Of course, autosave must be enabled and properly configured for this to happen.

**Reply by poundy** [reply]
In my mind, the autosave feature is to stop you attempting to start the next roast and losing your recorded  data from the previous roast in the rush to get ready to charge that next batch. It isn't intended to (automatically) save your incremental progress through a roast that is underway, as you have found... If you follow Dave's manual step, that might have helped, but certainly that isn't the common scenario that the software is designed to protect against without intervention.

Is the hottop (really) designed to stop completely when the "controlling software" no longer issues controls? That seems something that I wouldn't have expected, and I really expected that you'd still be able to manually navigate your way to a controlled end of roast much closer to the expected end parameters than just dumping at whatever point the failure happened.

**Reply by ProfStack** [reply]
Thanks roasterdave and poundy.  It appears there is no real solution for the saving of data if the battery is drained.  I was doing a second roast and more time elapsed between them than usual.  I'll have the charging cord ready as needed for future multiple roasting sessions.

As far as Hottop's stopping, when, in Artisan, the Control button is pressed during or right after a roast, the Hottop does indeed stop.  One must be quick to restart, dump, or at last resort pull the eject button in the rear of the roaster.
